src/boot/efi/pe.c

```diff
@@ -224,6 +253,69 @@ static void pe_locate_sections(
                 }
 }
 
+static bool pe_use_this_dtb(
+                const PeSectionHeader section_table[],
+                size_t n_section_table,
+                const void *base,
+                size_t dtb_offset,
+                size_t dtb_size,
+                size_t section_nb) {
+        EFI_STATUS err;
+
+        static const void *cached_base = NULL;
+        static const Device *cached_device = NULL;
+
+        const uint8_t *dtb = (const uint8_t *) base + dtb_offset;
+
+        err = devicetree_match(dtb, dtb_size);
+
+        /* Firmware does not provide the devicetree, so try matching against a list from .hwids section */
+        if (err == EFI_UNSUPPORTED) {
```

Comment:
reverse the check:

```c
if (err != EFI_UNSUPPORTED)
        return err;
```

and the rest of the function can be one level less indented.